src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder();
+        try {
+            LocalDate dateStart = LocalDate.parse(dateFrom, formatter);
+            LocalDate dateFinish = LocalDate.parse(dateTo, formatter);
+            int salary = 0;
+            builder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+
+            for (String name : names) {
+                for (String str : data) {
+                    String[] values = str.split(" ");
+                    LocalDate dateWork = LocalDate.parse(values[0], formatter);
+                    if (dateWork.isAfter(dateStart.minusDays(1))
+                            && dateWork.isBefore(dateFinish.plusDays(1))
+                            && name.equals(values[1])) {
+                        salary += Integer.parseInt(values[2]) * Integer.parseInt(values[3]);
+                    }
+                }
+                builder.append(System.lineSeparator()).append(name).append(" - ").append(salary);
+                salary = 0;
+            }
+        } catch (DateTimeParseException exc) {
+            System.out.printf("%s or %s is not parsable!%n", dateFrom, dateTo);
+            throw exc;
+        }
+        return builder.toString();
```

Comment:
I have a suggestion to improve the whole algorithm. Now you iterate names.length times over the data array. Which is bad, I suggest you try to implement an algorithm that iterates over the data array only 1 time, aggregates somewhere and somehow info, and then prints it. Because now on a Kate (if we talk about an example in the task description), you will iterate over all the data lines for the 3d time as a consequence parse each date and name again for the 3d time, this is insufficient.